### PR TITLE
feat: set MULTIPART_STRICT_ERROR value when mutipart fails to parse

### PR DESCRIPTION
**Pull request by fzipi**
## what

- set MULTIPART_STRICT_ERROR as it is used in coraza.conf-recommended but never set
- as the parser doesn't implement all the sub-variables from modsecurity, at least a global failure must be propagated

## why

- to catch possible evasion attempts with multipart data

### CODE REVIEW

**Comment by M4tteoP** [comment]
> as the parser doesn't implement all the sub-variables from modsecurity, at least a global failure must be propagated

Because of this, I would also make the rule output more aligned to Coraza removing all these variables that can confuse a user without the context of where this rule is coming from. 
https://github.com/corazawaf/coraza/blob/3b6497d5920f6f21f990cb77099458d83858029e/coraza.conf-recommended#L71-L85

**Comment by fzipi** [comment]
This definitely need to be cleaned up, only showing the variables supported... if any.

**Comment by fzipi** [comment]
We are also recommending this rule:
```
SecRule MULTIPART_UNMATCHED_BOUNDARY "@eq 1" \
    "id:'200004',phase:2,t:none,log,deny,msg:'Multipart parser detected a possible unmatched boundary.'"
```
But that variable is never set, so that rule cannot be triggered by any means.

**Comment by fzipi** [comment]
Updated the coraza.conf-recommended to a more realistic one.

File: coraza.conf-recommended [hunk 1]
```diff
@@ -230,17 +161,11 @@ SecAuditLogParts ABIJDEFHZ
...
-SecArgumentSeparator &
-
-# Settle on version 0 (zero) cookies, as that is what most applications
-# use. Using an incorrect cookie version may open your installation to
-# evasion attacks (against the rules that examine named cookies).
-#
-SecCookieFormat 0
+# SecCookieFormat 0
+# SecArgumentSeparator &
+# SecRule MULTIPART_UNMATCHED_BOUNDARY "@eq 1" \
```
Comment (M4tteoP) [inline coraza.conf-recommended:168]: Eventually for a follow-up PR, would be great to also have an engine test (under `testing/engine`) that triggers this rule, for an e2e-like test of the feature

**Review by M4tteoP — APPROVED** [review]
Looks good to me! Thanks 🙇🏻‍♂️

**Pull request merged**